Source/CBLDatabaseUpgrade.m

```diff
@@ -163,7 +163,70 @@ - (CBLStatus) moveAttachmentsDir {
             return CBLStatusFromNSError(error, 0);
         }
     }
-    return kCBLStatusOK;
+
+    return [self renameAttachmentFileNamesInDir: newAttachmentsPath];
+}
+
+
+- (CBLStatus) renameAttachmentFileNamesInDir: (NSString*)dir {
+    // CBL Android 1.0.4 and .NET 1.1.0 have lowercase attachment file names.
+    // This method will detect that and uppercase the file names if needed.
+    NSFileManager* fmgr = [NSFileManager defaultManager];
+
+    if (![fmgr fileExistsAtPath: dir])
+        return kCBLStatusOK;
+
+    NSError* error;
+    NSArray* content = [fmgr contentsOfDirectoryAtPath: dir error: &error];
+    if (error)
+        return CBLStatusFromNSError(error, kCBLStatusAttachmentError);
+
+    BOOL success = YES;
+    NSMutableArray* processed = [NSMutableArray array];
+    for (NSString* fileName in content) {
+        if (![[fileName pathExtension] isEqualToString:@"blob"])
+            continue;
+
+        NSString* filePath = [dir stringByAppendingPathComponent: fileName];
+        NSString* newPath = [self uppercaseAttachmentFileNameAtPath: filePath];
+
+        // Assume no both lowercase and uppercase filename mixed.
+        // Skip the renaming process if detecting that the file is already
+        // uppercased:
+        if ([filePath isEqualToString: newPath])
+            break;
+
+        if ([fmgr moveItemAtPath: filePath toPath: newPath error: &error]) {
+            [processed addObject: filePath];
+        } else {
+            Warn(@"Upgrade failed: Cannot rename attachment file %@: %@", filePath, error);
+            success = NO;
+            break;
+        }
+    }
+    
+    if (!success) {
+        // Backing out if there is an error found:
```

Comment:
Actually it depends. If the attachment folder was copied to the new location instead of moved, there is no need to backing that out.